multiload/src/load-graph.c

```diff
@@ -96,17 +96,13 @@ load_graph_draw (LoadGraph *g)
   {
     guint maxnet = 1;
     gint segments = 1;
-    gint combined;
     guint net_threshold;
 
     for (i = 0; i < g->draw_width; i++)
     {
       g->pos [i] = g->draw_height - 1;
-      combined = g->data[i][0] +
-                 g->data[i][1] +
-                 g->data[i][2];
-      if (combined > maxnet)
-        maxnet = combined;
+      if (g->data[i][3] > maxnet)
+        maxnet = g->data[i][3];
     }
```

Comment:
Build warning.
```
load-graph.c: In function 'load_graph_draw':

load-graph.c:104:25: warning: comparison of integer expressions of different signedness: 'gint' {aka 'int'} and 'guint' {aka 'unsigned int'} [-Wsign-compare]

  104 |       if (g->data[i][3] > maxnet)
```
https://travis-ci.org/github/mate-desktop/mate-applets/jobs/749808750#L12206
Possible it was already there.

Reply:
Yes, you're right. There are a lot of sign and width conversion errors.

libgtop mainly uses guint64 datatype for data, we use int for graph data, and cairo uses double datatype. And there are some unsigned longs in the middle... can use guint64 for graph data, although it may require a bit more memory.
```C
typedef void (*LoadGraphDataFunc) (int, int [], LoadGraph *);
```
```
multiload/src/load-graph.c:250:load_graph_unalloc (LoadGraph *g)
multiload/src/load-graph.c:280:load_graph_alloc (LoadGraph *g)
```

I also think some variables should be promoted to `gsize`, if they store a sizeof value or they are used for a malloc later.

Reply:
Sounds like another PR to fix it?

Reply:
Ok, I'm going to fix `guint net_threshold` and `*threshold*` in this PR, which is stored using 32-bit unsigned integers in gsettings and then stored in this variable as 16-bit unsigned integer. We need at least 30-bit uint = (log 1000000000 / (log 2) = 30
```
    <key name="netthreshold3" type="u">
      <range min="12" max="1000000000"/>
      <default>100000000</default>
      <summary>Network threshold 3 in bytes</summary>
    </key>
```

Reply:
@raveit65 done

Reply:
It's not ready to go, I detected an error in the network graph...

Reply:
@raveit65 done

Reply:
Warning is gone. thx.
What a huge commit...